### PR TITLE
Fixed typo in gz sim python api

### PR DESCRIPTION
**Pull request by ahcorde**
# 🦟 Bug fix

Fixes #<NUMBER>

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/gz/sim/Joint.cc [hunk 1]
```diff
@@ -94,7 +94,7 @@ void defineSimJoint(py::object module)
       py::arg("ecm"),
       py::arg("limits"),
       "Set the effort limits on a joint axis.")
-  .def("set_position_imits", &gz::sim::Joint::SetPositionLimits,
+  .def("set_position_limits", &gz::sim::Joint::SetPositionLimits,
```
Comment (arjo129) [inline python/src/gz/sim/Joint.cc:97]: Shouldn't we be deprecating this instead of removing it.

**Review by arjo129 — CHANGES_REQUESTED** [review]
I'm a bit torn by this change. On one hand given that we have released this, wouldn't this change break the API? Shouldn't we do a deprecation first?

On the other given how long its taken for this bug to be reported, maybe its fine to just go ahead and change the name.

File: python/src/gz/sim/Joint.cc (hunk 1 above)
Comment (azeey) [reply python/src/gz/sim/Joint.cc:97]: Yeah, I think we should deprecate the old function and add the new one.

Comment (ahcorde) [reply python/src/gz/sim/Joint.cc:112]: https://github.com/gazebosim/gz-sim/pull/2578/commits/27a46daeeef890f949514b88b086daf3eb8312ea

**Review by azeey — APPROVED** [review]
LGTM!

File: python/src/gz/sim/Joint.cc (hunk 1 above)
Comment (ahcorde) [reply python/src/gz/sim/Joint.cc:112]: @azeey this is wrong in older versions, what should we do?

**Pull request merged**